papergrid/src/util/string.rs

```diff
@@ -72,7 +52,7 @@ pub fn get_char_width(c: char) -> usize {
 
 /// Returns a string width (accouting all characters).
 pub fn get_string_width(text: &str) -> usize {
-    unicode_width::UnicodeWidthStr::width(text)
+    unicode_width::UnicodeWidthStr::width(text.replace(|c| c < ' ', "").as_str())
```

Comment:
Though now it became a heavy operation :(

Reply:
I wonder :thought_balloon: 

Maybe we rather must 'enforce' this in a way we did with 
`tabled::settings::formatting::Charset::clean`?
Just add a preprocessing step with this replacement and for those who are not sure about there sources it "assumed" to be used.
But for those who are pretty sure or did some processing themself we wouldn't waste this allocation? 

What I mean is famous - "why need to pay pay for what we are not using" (not sure how the quote goes exactly).

Reply:
Once I had an idea about a generic which would describe a table content as `Clean | Dirty` so we could do it out of the box.
And who are sure about there actions could use optimized version.

Reply:
I don't think the search portion of the replace operation is important to avoid; it's just a single search over the string contents, and when rendering text for human consumption, a character search won't add any significant overhead. The memory allocation, though, is worth avoiding if possible.

Ideally, there'd be a version of `str::replace` that returns a `Cow<str>`, and only allocates if it does any replacements. There isn't one in the standard library, but you could add one in `util::string`. That would eliminate the allocation overhead, and the overhead of a search for control characters isn't worth avoiding.

Reply:
I'll merge it, and do `Cow<str>`.
Will run benchmarks and look at it.

As I quite a bit favor the movement of it into its own setting :smile: 
*All though the one downside......we will need to recalculate it twice....... in most cases... but it's details.

Once again.
Thanks a lot.